src/backend/nodes/readfuncs.c

```diff
@@ -611,6 +613,9 @@ _readConst(void)
 		token = pg_strtok(&length); /* skip "<>" */
 	else
 		local_node->constvalue = readDatum(local_node->constbyval);
+	
+	if (local_node->constcollid == DEFAULT_COLLATION_OID)
+		local_node->constcollid = CLUSTER_COLLATION_OID();
```

Comment:
These questions are just for my knowledge.
1. Why do we need a special handling only for const and param? What about other functions such as `_readVar` or `_readAggref`?
2. Would you let me know corresponding test cases for both `_readConst` and `_readParam`?

Reply:
These functions are being used to depress the object definition, for example, pg_get_functiondef and tsql_get_functiondef

I added this code change as a test code change.. So I will remove this in next revision.